custom_components/frigate/views.py

```diff
@@ -208,9 +210,36 @@ def _create_path(self, path: str, **kwargs: Any) -> str | None:
             return f"api/events/{event_id}/clip.mp4"
         return None
 
-    def _permit_request(self, request: web.Request, config_entry: ConfigEntry) -> bool:
+    def _permit_request(self, request: web.Request, config_entry: ConfigEntry, **kwargs: Any) -> bool:
         """Determine whether to permit a request."""
-        return bool(config_entry.options.get(CONF_NOTIFICATION_PROXY_ENABLE, True))
+
+        is_notification_proxy_enabled = bool(config_entry.options.get(CONF_NOTIFICATION_PROXY_ENABLE, True))
+
+        # If proxy is disabled, immediately reject
+        if not is_notification_proxy_enabled:
+            return False
+
+        notification_expiration_mins = int(config_entry.options.get(CONF_EXPIRE_NOTIFICATIONS_AFTER_MINS, 0))
+
+        # If notification events never expire, immediately permit
+        if notification_expiration_mins == 0:
+            return True
+
+        try:
+            event_id_timestamp = int(kwargs["event_id"].partition(".")[0])
+            event_datetime = datetime.datetime.fromtimestamp(event_id_timestamp)
+            expiration_datetime = NotificationsProxyView._get_current_datetime() - datetime.timedelta(minutes=notification_expiration_mins)
```

Comment:
I think this would be clearer to take `event_datetime` + `notification_expiration_mins`, then compare now against that expiration datetime.

Reply:
It sounds like the same check. 
`A >= B - C`
vs
`A + C >= B`

If you feel strongly it would aid readability, I can change it.

Reply:
Sorry, yeah, it's readability thing not an functionality thing. Took me a few reads to untangle this, but may just be me -- feels a bit more intuitive to just compare the (event_datetime + expire) vs the current time?